sdk/data/azcosmos/queryengine/cosmos_query_engine.go

```diff
@@ -13,15 +13,36 @@ type QueryEngine interface {
 type QueryRequest struct {
 	// PartitionKeyRangeID is the ID of the partition key range from which data is requested.
 	PartitionKeyRangeID string
+	// The index of this request, within the partition key range.
+	//
+	// This value will always increase for subsequent requests for the same partition key range.
+	// It must be provided back to the pipeline when providing data, so that the pipeline can ensure that data is provided in order.
+	Index uint
 	// Continuation is the continuation token to use in the request.
 	Continuation string
+	// The query to execute for this partition key range, if different from the original query.
+	Query string
+	// If a query is specified, this flag indicates if the query parameters should be included with that query.
+	//
+	// Sometimes, when an override query is specified, it differs in structure from the original query, and the original parameters are not valid.
+	IncludeParameters bool
+	// If specified, indicates that the SDK should IMMEDIATELY drain all remaining results from this partition key range, following continuation tokens, until no more results are available.
+	// All the data from this partition key range should be provided BEFORE any new items will be made available.
```

Comment:
Just a small addition to clarify the interaction between RequestId and Drain.

```suggestion
	// All the data from this partition key range should be provided BEFORE any new items will be made available.
	// The data may be provided in multiple QueryResults, but every result correlated to this request should have the same RequestId value.
```